dts/bindings/mtd/zephyr,smbus-eeprom.yaml

```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2022,2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: Zephyr SMBus EEPROM binding
+
+compatible: "zephyr,smbus-eeprom"
```

Comment:
Why the `zephyr,` prefix? Don't these devices represent actual hardware?

Reply:
This is driver for Qemu EEPROM "kind of" devices, I used it for testing SMBus only.
Source:
https://github.com/zephyrproject-rtos/qemu/blob/2cc2e86de6e0e4316265d34e8e935afade8c422e/hw/i386/pc_q35.c#L315
https://github.com/zephyrproject-rtos/qemu/blob/zephyr-qemu-v7.0.0/hw/i2c/smbus_eeprom.c

Reply:
@henrikbrixandersen using heavy macrobatic I can make the AT24 EEPROM driver even more unreadable including SMBus part, the current code heavily uses `#idef`s to separate SPI and I2C parts.